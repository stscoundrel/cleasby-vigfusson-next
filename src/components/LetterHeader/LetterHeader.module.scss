.section {
  margin: 20px 0 30px 0;
}

.subHeading {
  display: block;
  margin-bottom: 30px;
}

.title {
  margin-bottom: 0;
}

.count {
  display: block;
  margin-bottom: 1rem;
}