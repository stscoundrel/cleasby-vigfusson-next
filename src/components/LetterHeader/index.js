import styles from './LetterHeader.module.scss'

export default function LetterLink({ letter, count }) {
  return (
   <header className={styles.section}>
    <h1 className={styles.title}>Letter {letter.toUpperCase()}</h1>
    <small className={styles.subHeading}>
      Cleasby & Vigfusson Dictionary - Letter {letter.toUpperCase()}
    </small>
    <p>Old Norse words starting with letter {letter.toUpperCase()}</p>
    <small className={styles.count}>Total of {count} words</small>
  </header>
  )
}
