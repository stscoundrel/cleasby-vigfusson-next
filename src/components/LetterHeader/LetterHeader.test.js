import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import LetterHeader from './index'

describe('LetterHeader component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<LetterHeader letter="a" count={12345} />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<LetterHeader letter="s" count={60} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Outputs correct count', () => {
    const tree = renderer.create(<LetterHeader letter="s" count={60} />)
    const { root } = tree
    expect(root.findAllByType('small')[1].children.join(' ').includes('60')).toBeTruthy()
  })
})
