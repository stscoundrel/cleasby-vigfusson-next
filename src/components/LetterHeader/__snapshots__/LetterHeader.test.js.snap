// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterHeader component Matches snapshot 1`] = `
<header
  className="section"
>
  <h1
    className="title"
  >
    Letter 
    S
  </h1>
  <small
    className="subHeading"
  >
    Cleasby & Vigfusson Dictionary - Letter 
    S
  </small>
  <p>
    Old Norse words starting with letter 
    S
  </p>
  <small
    className="count"
  >
    Total of 
    60
     words
  </small>
</header>
`;
